<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/emerito/catkin_ws/src/donatello_sim/urdf/donatello.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!-- TO-DO -->
<!-- change sizes -->
<!-- change inertia -->
<!-- MEDIDAS EN DECIMETROS -->
<robot name="donatello">
  <!-- SIZES -->
  <!-- INERTIA -->
  <!-- LINKS -->
  <link name="dummy_link">
    </link>

 <!-- Base Blanca -->
  <link name="base_link">
    <visual>
      <geometry>
        <box size="5.0 3.48 1.65"/> <!-- Geometría. x y z (altura). -->
      </geometry>
    </visual>
    <collision>
      <geometry>
        <box size="5.0 3.48 1.65 "/> <!-- Colisión. x y z (altura). -->
      </geometry>
    </collision>

    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.083" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.1"/>
    </inertial>
  </link>
  <!-- Inercia No revisado -->

 <!-- Link trasero -->
  <link name="rear_axle_link">
    <visual>
      <geometry>
        <cylinder length="0.3" radius="0.06"/>  <!-- Geometría del cilíndro -->
      </geometry>
      <origin rpy="0 -1.5707963 0" xyz="-1 0 0"/> <!-- Origen. rpy: lugar de uniones. xyz: espacio. -->
    </visual>

    <visual>
      <geometry>
        <box size="0.3 3.5 0.3"/>   <!-- Perfil horizontal -->
      </geometry>
      <origin rpy="0 0 0" xyz="-0.07500000000000001 0 0"/>
    </visual>

    <visual>
      <geometry>
        <box size="0.3 0.3 1.5"/> <!-- Perfil vertical izquierdo -->
      </geometry>
      <origin rpy="0 0 0" xyz="-0.07500000000000001 0.325 -0.17500000000000002"/>
    </visual>

    <visual>
      <geometry>
        <box size="0.3 0.3 1.5"/> <!-- Perfil vertical derecho -->
      </geometry>
      <origin rpy="0 0 0" xyz="-0.07500000000000001 -0.325 -0.17500000000000002"/>
    </visual>
    <!-- 24/10/24 -->

    <collision>
      <geometry>
        <cylinder length="0.05" radius="0.025"/>
      </geometry>
      <origin rpy="0 -1.5707963 0" xyz="-0.025 0 0"/>
    </collision>
    <collision>
      <geometry>
        <box size="0.05 0.6 0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.07500000000000001 0 0"/>
    </collision>
    <collision>
      <geometry>
        <box size="0.05 0.05 0.4"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.07500000000000001 0.325 -0.17500000000000002"/>
    </collision>
    <collision>
      <geometry>
        <box size="0.05 0.05 0.4"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.07500000000000001 -0.325 -0.17500000000000002"/>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.083" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.1"/>
    </inertial>
  </link>

  <link name="front_left_axle_link">
    <visual>
      <geometry>
        <box size="0.05 0.05 0.5656854240000001"/>
      </geometry>
      <origin rpy="0 0.78539815 0" xyz="-0.2 0 -0.2"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.05 0.05 0.5656854240000001"/>
      </geometry>
      <origin rpy="0 -0.78539815 0" xyz="0.2 0 -0.2"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.05 0.05 0.5656854240000001"/>
      </geometry>
      <origin rpy="0 0.78539815 0" xyz="-0.2 0 -0.2"/>
    </collision>
    <collision>
      <geometry>
        <box size="0.05 0.05 0.5656854240000001"/>
      </geometry>
      <origin rpy="0 -0.78539815 0" xyz="0.2 0 -0.2"/>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.083" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.1"/>
    </inertial>
  </link>
  <link name="front_right_axle_link">
    <visual>
      <geometry>
        <box size="0.05 0.05 0.5656854240000001"/>
      </geometry>
      <origin rpy="0 0.78539815 0" xyz="-0.2 0 -0.2"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.05 0.05 0.5656854240000001"/>
      </geometry>
      <origin rpy="0 -0.78539815 0" xyz="0.2 0 -0.2"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.05 0.05 0.5656854240000001"/>
      </geometry>
      <origin rpy="0 0.78539815 0" xyz="-0.2 0 -0.2"/>
    </collision>
    <collision>
      <geometry>
        <box size="0.05 0.05 0.5656854240000001"/>
      </geometry>
      <origin rpy="0 -0.78539815 0" xyz="0.2 0 -0.2"/>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.083" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.1"/>
    </inertial>
  </link>
  <!-- JOINTS -->
  <joint name="dummy_joint" type="fixed">
    <parent link="dummy_link"/>
    <child link="base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <joint name="rear_axle_joint" type="continuous">
    <parent link="base_link"/>
    <child link="rear_axle_link"/>
    <origin rpy="0 0 0" xyz="-0.5 0 -0.15"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="front_left_axle_joint" type="continuous">
    <parent link="base_link"/>
    <child link="front_left_axle_link"/>
    <origin rpy="0 0 0" xyz="0.3 0.325 -0.15"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="front_right_axle_joint" type="continuous">
    <parent link="base_link"/>
    <child link="front_right_axle_link"/>
    <origin rpy="0 0 0" xyz="0.3 -0.325 -0.15"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="back_left_wheel_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.083" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.1"/>
    </inertial>
  </link>
  <joint name="back_left_wheel_joint" type="continuous">
    <parent link="rear_axle_link"/>
    <child link="back_left_wheel_link"/>
    <origin rpy="1.5707963 0 0" xyz="-0.07500000000000001 0.39999999999999997 -0.4"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="back_right_wheel_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.083" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.1"/>
    </inertial>
  </link>
  <joint name="back_right_wheel_joint" type="continuous">
    <parent link="rear_axle_link"/>
    <child link="back_right_wheel_link"/>
    <origin rpy="-1.5707963 0 0" xyz="-0.07500000000000001 -0.39999999999999997 -0.4"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="center_left_wheel_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.083" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.1"/>
    </inertial>
  </link>
  <joint name="center_left_wheel_joint" type="continuous">
    <parent link="front_left_axle_link"/>
    <child link="center_left_wheel_link"/>
    <origin rpy="1.5707963 0 0" xyz="-0.4 0.07500000000000001 -0.4"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="center_right_wheel_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.083" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.1"/>
    </inertial>
  </link>
  <joint name="center_right_wheel_joint" type="continuous">
    <parent link="front_right_axle_link"/>
    <child link="center_right_wheel_link"/>
    <origin rpy="-1.5707963 0 0" xyz="-0.4 -0.07500000000000001 -0.4"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="front_left_wheel_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.083" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.1"/>
    </inertial>
  </link>
  <joint name="front_left_wheel_joint" type="continuous">
    <parent link="front_left_axle_link"/>
    <child link="front_left_wheel_link"/>
    <origin rpy="1.5707963 0 0" xyz="0.4 0.07500000000000001 -0.4"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="front_right_wheel_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.083" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.1"/>
    </inertial>
  </link>
  <joint name="front_right_wheel_joint" type="continuous">
    <parent link="front_right_axle_link"/>
    <child link="front_right_wheel_link"/>
    <origin rpy="-1.5707963 0 0" xyz="0.4 -0.07500000000000001 -0.4"/>
    <axis xyz="0 0 1"/>
  </joint>
</robot>
<!-- 
EJECUTAR ESTA LINEA ANTES DE HACER catkin_make

rosrun xacro xacro ~/catkin_ws/src/donatello_sim/urdf/donatello.urdf.xacro > ~/catkin_ws/src/donatello_sim/urdf/donatello.urdf

check_urdf ~/catkin_ws/src/donatello_sim/urdf/donatello.urdf

-->

